### PR TITLE
edit a todo

### PR DESCRIPTION
**Pull request by excellencetechnologies**

### CODE REVIEW

File: index.jsx [hunk 1]
```diff
@@ -35,22 +40,51 @@ export class Todo extends React.Component {
...
         new_todo_lists.push( { text : newtodo, marked : false, id : new_key } );
-        
         this.setState({ todos : new_todo_lists });
-
+    }
+    editToDo( todo_edit ){
         
+        if( todo_edit ){
+            let x = 0;
+            this.state.todos.forEach( function(p){
```
Comment (upexcel) [inline index.jsx]: use filter or reduce instead of forEach

File: index.jsx [hunk 2]
```diff
@@ -35,22 +40,51 @@ export class Todo extends React.Component {
...
+            
+            if( x != 0 ){
+                this.setState({ edit_todo_id : x.id, edit_todo_text : x.text });
+            }
+        }
+    }
+    updatetodo( id, new_text ){
+        let new_todo_lists = [];
+        if( id ){
+            this.state.todos.forEach( function(p){
```
Comment (upexcel) [inline index.jsx]: forEach is rarely used. see other function relevent to what you want to do like map, filter, reduce etc

File: index.jsx [hunk 3]
```diff
@@ -35,22 +40,51 @@ export class Todo extends React.Component {
...
+                if( p.id == id ){
+                    p.text = new_text;
+                }
+                new_todo_lists.push( p );
```
Comment (upexcel) [inline index.jsx]: if u use proper function like map u dont need to have a new_todo_lists

File: todo/addtodo.jsx [hunk 4]
```diff
@@ -4,7 +4,10 @@ import classNames from 'classnames';
 export class AddToDo extends React.Component{
     constructor( props ){
         super( props );
-        this.state = { newtodo : '', is_form_empty : 0 };
+        this.state = { 
+            newtodo : '', 
+            is_form_empty : 0
```
Comment (upexcel) [inline todo/addtodo.jsx]: do we need two state variables?
its important in react to have minimum state

File: todo/addtodo.jsx [hunk 5]
```diff
@@ -32,8 +32,8 @@ export class AddToDo extends React.Component{
         return(
             <form onSubmit={this.addtodo} className="todoForm">
                 <div className={inputClass}>
-                    <label htmlFor="todo">Add Todo</label>
-                    <input onChange={this.handleChange} type="text" className="form-control" id="todo" placeholder="Your Todo.." value={this.state.newtodo }></input>
+                    <label htmlFor="todo">Add ToDO</label>
+                    <input onChange={this.handleChange} type="text" className="form-control" id="todo" placeholder="Your Todo.." value={ this.state.newtodo }></input>
```
Comment (upexcel) [inline todo/addtodo.jsx]: use refs in input. its useful in long term

File: todo/edittodo.jsx [hunk 6]
```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+import classNames from 'classnames';
+
+export class EditToDo extends React.Component{
```
Comment (upexcel) [inline todo/edittodo.jsx]: why two components required for Add and Edit?
 can we not manage with one using if conditions

File: index.jsx (hunk 1 above)
Comment (excellencetechnologies) [inline index.jsx]: done with filter

File: index.jsx (hunk 2 above)
Comment (excellencetechnologies) [inline index.jsx]: done with map.

File: index.jsx (hunk 3 above)
Comment (excellencetechnologies) [inline index.jsx]: yes, solved with map function

File: todo/addtodo.jsx (hunk 4 above)
Comment (excellencetechnologies) [inline todo/addtodo.jsx]: yes 2nd state is required, with out this form textbox shows required validation even without submission

File: todo/addtodo.jsx (hunk 5 above)
Comment (excellencetechnologies) [inline todo/addtodo.jsx]: not able to get this ref yet.

File: todo/edittodo.jsx (hunk 6 above)
Comment (excellencetechnologies) [inline todo/edittodo.jsx]: Earlier i tried with the same thing, but not able to get success in this, so did with seaprate components.

**Pull request merged**